### PR TITLE
Deprecate Hanami::Interactor::Result#failing?

### LINKED ISSUE #172: Interactor uses gramatically wrong terms for success/failure
Thank you for your work on this project!

When an Interactor call was a success, we check with either

```ruby
result = Signup.new().call
result.success?
# OR
result.successful?
```

but we check failure with

```ruby
result.failing?
```

This doesn't make grammatic sense, because the opposite of `success?` would be `failure?` and the opposite of `successful?` would still be `failure?`.  However, `failing?` is definitely not correct, because its opposite would be `succeeding?`.

I think this is quite confusing, even though it is a small nitpick.  Maybe `failing?` could be deprecated and aliased to `failure?`?

Thoughts? Thanks again!

### PR DESCRIPTION
**Pull request by mgrachev**
Closes #172

### CODE REVIEW

File: lib/hanami/interactor.rb [hunk 1]
```diff
@@ -57,7 +59,19 @@ def successful?
...
       def failing?
+        Utils::Deprecation.new("`Hanami::Interactor::Result#failing?' is deprecated, please use `Hanami::Interactor::Result#failure?'")
+        failure?
+      end
+
+      # Check if the current status is not successful
+      #
+      # @return [TrueClass,FalseClass] the result of the check
+      #
+      # @since 0.9.2
```
Comment (AlfonsoUceda) [inline lib/hanami/interactor.rb]: @mgrachev  could you put to x.x.x?

**Comment by TiteiKo** [comment]
Looks good to me 👍

File: lib/hanami/interactor.rb (hunk 1 above)
Comment (mgrachev) [inline lib/hanami/interactor.rb]: Done.

**Comment by cllns** [comment]
Thanks @mgrachev!

**Pull request merged**